pkg/api/api.go

```diff
@@ -237,47 +242,47 @@ func Series(ctx context.Context, client promapi.Client, matches []string, startT
 	q.Set("start", formatTime(startTime))
 	q.Set("end", formatTime(endTime))
 
-	_, body, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
+	resp, body, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
 	if err != nil {
-		return nil, warnings, err
+		return nil, resp.StatusCode, warnings, err
 	}
 
 	var mset []model.LabelSet
-	return mset, warnings, json.Unmarshal(body, &mset)
+	return mset, resp.StatusCode, warnings, json.Unmarshal(body, &mset)
 }
 
 func LabelNames(ctx context.Context, client promapi.Client, startTime time.Time, endTime time.Time,
-	cache bool) ([]string, promapiv1.Warnings, error) {
+	cache bool) ([]string, int, promapiv1.Warnings, error) {
 	u := client.URL(epLabels, nil)
 	q := u.Query()
 	q.Set("start", formatTime(startTime))
 	q.Set("end", formatTime(endTime))
 
 	u.RawQuery = q.Encode()
 
-	_, body, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
+	resp, body, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
 	if err != nil {
-		return nil, warnings, err
+		return nil, resp.StatusCode, warnings, err
```

Comment:
Shouldn't we do `resp == nil` check here as well?

Reply:
👍